### DISCUSSION TITLE
Настройка времени

### DISCUSSION THREAD

**Discussion by SkolExChita** [vvip-68/LedPanelWiFi#96] [Вопросы и ответы] [answered]
Обратил внимание что время показывает московское (UTC+3) настройки в web менял, в прошивке менял, всё равно выдаёт московское. при загрузке до ntp загружается, в мониторе порта следующее:
20:54:50.637 -> TIME: Получено время NTP: 2024-12-15T11:53
20:54:50.905 -> Погода получена: OpenWeatherMap
20:54:50.905 -> Город: Чита
20:54:50.955 -> Сейчас: ясно, -21ºC

Можете что нибудь посоветовать?

**Comment by vvip-68** [comment]
Нет, ничего не могу. Часовой пояс выбирается в настройках часов. Какой выбрали - тот и должен показываться. Можно выбрать не город, а смещение UTC, при желании. Погода к часовому поясу никакого отношения не имеет.

Проверил - при выборе часового пояса читы часы показывают сейчас 22:54. В Москве сейчас 16:54

**Reply by SkolExChita** [reply]
У меня, ещё со старых времён, через розетку с таймером. При каждом включении всё сбрасываеться (
Буду реле прикручивать...
 Но опять же при отключении света всё опять слетит...  
Если не трудно, попробуйте перезагрузить с другим часовым поясом, может тогда что нибудь всплывёт?...

**Reply by vvip-68** [reply, marked as answer]
Сочувствую, но ничем  помочь ни могу. Кроме вас ни у кого таких проблем не возникало (никто не писал). У меня тоже такого не наблюдается. Ничего не сбрасывается ни при перезагрузке, ни при полном выключении и включении потом. Все настройки хранятся в энергонезависимой памяти микроконтроллера.

**Reply by SkolExChita** [reply]
Хорошо, буду думать)

**Reply by SkolExChita** [reply]
отредактировал скетч timeprocessor

`#include "a_def_hard.h"                 // нам нужны макросы для дебага


#ifndef TZONE
#define TZONE PSTR("GMT-9")              // default Time-Zone
#endif

extern bool vDEBUG_SERIAL;              // внешняя глобальная переменная включения дебага

// stub zone for a  <+-nn> names
static const char P_LOC[] PROGMEM = "LOC";

// static member must be defined outside the class
callback_function_t TimeProcessor::timecb = nullptr;
`
После этого всё заработало)))

**Reply by vvip-68** [reply]
Хм... Почему же у меня и у других без этих изменений всё работает?
Ну ладно - отредактировали, заработало - и хорошо...
